aws/rust-runtime/aws-config/src/imds/credentials.rs

```diff
@@ -192,21 +192,20 @@ impl ImdsCredentialsProvider {
     //
     // This allows continued use of the credentials even when IMDS returns expired ones.
     fn maybe_extend_expiration(&self, expiration: SystemTime) -> SystemTime {
+        let now = self.time_source.now();
+        // If credentials from IMDS are not stale, use them as they are.
+        if now < expiration {
+            return expiration;
+        }
+
         let rng = fastrand::Rng::with_seed(
-            self.time_source
-                .now()
-                .duration_since(SystemTime::UNIX_EPOCH)
+            now.duration_since(SystemTime::UNIX_EPOCH)
                 .expect("now should be after UNIX EPOCH")
                 .as_secs(),
         );
         // calculate credentials' refresh offset with jitter
-        let refresh_offset =
-            CREDENTIAL_EXPIRATION_INTERVAL + Duration::from_secs(rng.u64(120..=600));
-        let new_expiry = self.time_source.now() + refresh_offset;
-
-        if new_expiry < expiration {
-            return expiration;
-        }
+        let refresh_offset = CREDENTIAL_EXPIRATION_INTERVAL + Duration::from_secs(rng.u64(0..=300));
```

Comment:
Per our spec, 15-minute is the minimum amount of time credentials are valid for. Setting `refresh_offset` to something longer than that may have the risk of the credentials expiring before the next refresh (especially when the credentials expiry is set to the very minimum of 15 minutes from now).

Now that we've added the early return, however, it may not matter much because valid credentials will be used as they are without the extension. But just wanted to make this inline with the spec.

Added comment in c265e86.